Plane: remove chan params

IamPete1 opened the pull request:
This removes the remaining `CHAN` params in plane in favor of `RCx_OPTION`.  (except `OVERRIDE_CHAN`) 

Should be NFC.

Fixes #16881

WickedShell commented on ArduPlane/Parameters.h:
```diff
@@ -119,13 +119,13 @@ class Parameters {
         k_param_flap_slewrate,
         k_param_rtl_autoland,
-        k_param_override_channel,
+        k_param_override_channel, // unused
```
Doesn't look unused to me?

IamPete1 replied:
![image](https://user-images.githubusercontent.com/33176108/111006122-19667980-8384-11eb-8336-8dcd0c1ef473.png)

Unless I'm failing at VSCode.....

IamPete1 replied:
Ah, I think i'm being caught out by the macro stuff, your right.

WickedShell reviewed (changes requested):
Override channel definitely needs an option of not moving as I use it quite a lot with the mode switch, not sure if any of the others are commonly used that way.

Hwurzburg commented on ArduPlane/Parameters.cpp:
```diff
@@ -650,12 +644,6 @@ const AP_Param::Info Plane::var_info[] = {
     // @User: Advanced
     GSCALAR(reset_switch_chan,      "RST_SWITCH_CH",  0),
```
why not this one too?

IamPete1 replied:
because I did not find it, ;) I will add it too.

IamPete1 replied:
done

IamPete1 commented:
I found another one, `Q_TAILSIT_MASKCH`, but I think it can wait for another PR

tridge commented on ArduPlane/Parameters.cpp:
```diff
@@ -1339,6 +1315,26 @@ const AP_Param::ConversionInfo conversion_table[] = {
+    RC_Channel::AUX_FUNC::PARACHUTE_RELEASE,
+    RC_Channel::AUX_FUNC::FBWA_TAILDRAGGER,
+    RC_Channel::AUX_FUNC::MODE_SWITCH_RESET,
+};
```
static assert for 2 tables of same length

tridge replied:
even better as one table

IamPete1 replied:
fixed

tridge commented on ArduPlane/Parameters.cpp:
```diff
@@ -1339,6 +1315,26 @@ const AP_Param::ConversionInfo conversion_table[] = {
     { Parameters::k_param_fence_autoenable,   0,      AP_PARAM_INT8, "FENCE_AUTOENABLE"},
 };
 
+const AP_Param::ConversionInfo rc_chan_params[] = {
```
should be static

IamPete1 replied:
done

IamPete1 commented:
rebased, fixed issues and re-tested. 

Note that FBWA taildrager spams messages on and off messages when the siwtch is high. But it does that on master too.

![image](https://user-images.githubusercontent.com/33176108/111322147-8322a400-8660-11eb-8bf4-164a5410cb48.png)
